### PR TITLE
Release 17.1.0

### PR DESCRIPTION
**Pull request by doggydogworld**
* Updated golang.org/x/net to v0.33.0 (addresses CVE-2024-45338). [#50397](https://github.com/gravitational/teleport/pull/50397)
* Prevents panic in SAML app or OIDC integration deletion when a plugin type is not `PluginTypeAWSIdentityCenter` or `PluginAWSICSettings` is nil. [#50360](https://github.com/gravitational/teleport/pull/50360)
* Fix missing roles in access lists causing users to be locked out of their account. [#50298](https://github.com/gravitational/teleport/pull/50298)
* Added support for connecting to PostgreSQL databases using WebUI. [#50287](https://github.com/gravitational/teleport/pull/50287)
* Improved the performance of Teleport agents serving a large number of resources in Kubernetes. [#50279](https://github.com/gravitational/teleport/pull/50279)
* Improve performance of Kubernetes App Auto Discover. [#50269](https://github.com/gravitational/teleport/pull/50269)
* Added more granular access controls for SSH port forwarding. Access to remote or local port forwarding can now be controlled individually using the new `ssh_port_forwarding` role option. [#50241](https://github.com/gravitational/teleport/pull/50241)
* Properly close ssh port forwarding connections to prevent requests hanging indefinitely. [#50238](https://github.com/gravitational/teleport/pull/50238)
* Teleport's RDP client now sets the load balancing cookie to improve compatibility with local traffic managers. [#50226](https://github.com/gravitational/teleport/pull/50226)
* Fixes an intermittent EKS authentication failure when dealing with EKS auto-discovery. [#50197](https://github.com/gravitational/teleport/pull/50197)
* Expose /.well-known/jwks-okta public endpoint for Okta API services type App. [#50177](https://github.com/gravitational/teleport/pull/50177)
* Switched to a new role editor UI. [#50030](https://github.com/gravitational/teleport/pull/50030)
* Added support for multiple ports to TCP applications. [#49711](https://github.com/gravitational/teleport/pull/49711)

Enterprise:
* Adds AWS Account name to Identity Center Roles and resources. Some manual cleanup may be required where users and Access Lists have been assigned the obsolete roles.

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

**Comment by tigrato** [comment]
Can you wait for https://github.com/gravitational/teleport/pull/50386?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Changelog
 
+## 17.1.0 (12/18/24)
```
Comment (r0mant) [inline CHANGELOG.md:3]: We should first include a section with new features in this minor release: https://goteleport.com/docs/upcoming-releases/#1710.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+
+### Access requests support for AWS Identity Center
+
+AWS Identity Center integration will allow users to request short or long term access to permission sets via Access Requests.
```
Comment (r0mant) [inline CHANGELOG.md:7]: Should change all "will allow" to "now allows" :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+* Added more granular access controls for SSH port forwarding. Access to remote or local port forwarding can now be controlled individually using the new `ssh_port_forwarding` role option. [#50241](https://github.com/gravitational/teleport/pull/50241)
+* Properly close ssh port forwarding connections to prevent requests hanging indefinitely. [#50238](https://github.com/gravitational/teleport/pull/50238)
+* Teleport's RDP client now sets the load balancing cookie to improve compatibility with local traffic managers. [#50226](https://github.com/gravitational/teleport/pull/50226)
+* Fixes an intermittent EKS authentication failure when dealing with EKS auto-discovery. [#50197](https://github.com/gravitational/teleport/pull/50197)
+* Expose /.well-known/jwks-okta public endpoint for Okta API services type App. [#50177](https://github.com/gravitational/teleport/pull/50177)
+* Switched to a new role editor UI. [#50030](https://github.com/gravitational/teleport/pull/50030)
+* Added support for multiple ports to TCP applications. [#49711](https://github.com/gravitational/teleport/pull/49711)
```
Comment (tigrato) [inline CHANGELOG.md:43]:
```suggestion
* Added support for multiple ports to TCP applications. [#49711](https://github.com/gravitational/teleport/pull/49711)
* Allow multiple consecutive occurrences of `-` and `.` in SSH server hostnames.  [#50410](https://github.com/gravitational/teleport/pull/50410)
```

**Review by r0mant — DISMISSED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (r0mant) [reply CHANGELOG.md]: @doggydogworld You changed it here but not anywhere below. Those still say "will support ..."

**Review by fheinecke — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**